src/arrayMethodJoin.js

```diff
@@ -5,7 +5,63 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
```

Comment:
```suggestion
  [].__proto__.join2 = function(separator = ',') {
```
You can set the default separator. You can remove most of checks in that case. It will make your code much more cleaner)